### ISSUE TITLE
Have battery status of devices passed through to Home Assistant

### ISSUE THREAD

**Issue by deepcoder** [MartinPham/FindMySync#41]
I see a JSON pair "batteryStatus" : {0,1,2...?} in the Items.data file for devices, however this info does not seem to be passed to Home Assistant. Is there a way to pass this data?

Thanks!

**Comment by ghost** [comment]
It is passed to home assistant, this is using a deprecated API, but you should see a "battery" attribute on the entity.

**Comment by miguelangel-nubla** [comment]
@ext4xfs  I don't see it, could you provide more details about the API it is supposedly using?

**Comment by ghost** [comment]
get battery level
https://github.com/MartinPham/FindMySync/blob/842bb6454b2796b4ccc332ffe3742eeabf9cbd70/FindMySync/Synchronizer.swift#L640-L654

entities to update
https://github.com/MartinPham/FindMySync/blob/842bb6454b2796b4ccc332ffe3742eeabf9cbd70/FindMySync/Synchronizer.swift#L656-L665

request body sent to home assistant
https://github.com/MartinPham/FindMySync/blob/842bb6454b2796b4ccc332ffe3742eeabf9cbd70/FindMySync/Synchronizer.swift#L740-L756

device_tracker domain
https://www.home-assistant.io/integrations/device_tracker/#device_trackersee-action

rest api `/api/services/<domain>/<service>`
https://developers.home-assistant.io/docs/api/rest/#actions

**Comment by miguelangel-nubla** [comment]
Thank you for the detailed response.

I think we are talking about different things, the code is reading `batteryLevel` which I only see on `Devices.data` for the full devices (iphone, etc).

This issue is about getting the battery status of airtags and similar devices in `Items.data`, which seem to only have a `batteryStatus` attribute.

On my devices `batteryStatus=1` corresponds to a full battery, `batteryStatus=2` next to full.
I don't know if that attribute is supposed to represent battery level but it is certainly not parsed or sent to home assistant right now.

**Comment by ghost** [comment]
ah I see what you mean, yeah it's there for devices but not items.

**Comment by miguelangel-nubla** [comment]
@MartinPham any chance to include battery status for items?

**Comment by jasgroupit** [comment]
yeah need battery status for items its quite important so we know the status of the items battery so even if dont have iphone near us we know battery conditions
@MartinPham

**Comment by ghost** [comment]
I don't think it's useful information, batteryStatus 1 or 2 doesn't tell you much for airtags at least, since they use a coin battery without any fancy circuitry.

**Comment by jasgroupit** [comment]
At the very least using battery status 1 or 2 we can tell if the battery is in ok or degraded stage so we can monitor , if u can add that or enable that would great so i can scene in homeassistant to cross check if 24 hour

**Comment by jasgroupit** [comment]
@MartinPham @ext4xfs the status for items possible ?

**Comment by manonstreet** [comment]
FWIW, my fork [FindMySyncPlus](https://github.com/manonstreet/FindMySyncPlus) supports this capability.  But it was mainly written to support new macOS's versions which have the data files encrypted.  My function can be ported to this project if desired:

```
private func batteryFromStatus(_ any: Any?) -> Double? {
    // Accept a few common shapes for Items:
    // - String statuses like "full", "normal", "medium", "low", "verylow", "critical"
    // - Numeric percentages (0-100)
    // - Fractions (0.0-1.0)
    if let d = any as? Double {
        if d.isNaN || d.isInfinite { return nil }
        return d > 1.0 ? max(0.0, min(1.0, d / 100.0)) : max(0.0, min(1.0, d))
    }
    if let i = any as? Int {
        return max(0.0, min(1.0, Double(i) / 100.0))
    }
    if let s = (any as? String)?.trimmingCharacters(in: .whitespacesAndNewlines).lowercased() {
        switch s {
        case "full", "max", "high":
            return 1.0
        case "normal", "good":
            return 0.75
        case "medium", "fair":
            return 0.5
        case "low":
            return 0.25
        case "verylow", "critical", "very low":
            return 0.1
        default:
            // Try to parse numeric-in-string (either percent or fraction)
            if let asNum = Double(s) {
                return asNum > 1.0 ? max(0.0, min(1.0, asNum / 100.0)) : max(0.0, min(1.0, asNum))
            }
            return nil
        }
    }
    return nil
}
```
